src/points/saga.ts

```diff
@@ -0,0 +1,82 @@
+import { call, put, select, spawn, takeLeading } from 'typed-redux-saga'
+import {
+  getInitialHistoryStarted,
+  getMoreHistoryStarted,
+  getHistorySucceeded,
+  getHistoryError,
+} from 'src/points/slice'
+import { safely } from 'src/utils/safely'
+import { retrieveSignedMessage } from 'src/pincode/authentication'
+import networkConfig from 'src/web3/networkConfig'
+import { GetHistoryResponse } from 'src/points/types'
+import { fetchWithTimeout } from 'src/utils/fetchWithTimeout'
+import { walletAddressSelector } from 'src/web3/selectors'
+import { getPointsHistoryNextPageUrlSelector } from 'src/points/selectors'
+import Logger from 'src/utils/Logger'
+
+const TAG = 'Points/saga'
+
+export async function fetchHistory(address: string, url?: string): Promise<GetHistoryResponse> {
+  const signedMessage = await retrieveSignedMessage()
+  const response = await fetchWithTimeout(url ?? networkConfig.getPointsHistoryUrl, {
+    method: 'GET',
+    headers: {
+      authorization: `Valora ${address}:${signedMessage}`,
```

Comment:
curious, why we need authentication on this endpoint?

Reply:
Hmm, yeah I suppose we don't, though it means we would need to add another query param for `address`

Reply:
(updated the backend to reflect this!)